### PR TITLE
UAR-894: fix reloading of BO/MO data during no change journey

### PR DESCRIPTION
**Pull request by n-r-b**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-894

### Change description

* Updated the resetting of the appData for BO/MOs when 'no change' is selected on the ‘Do you need to make any changes to this overseas entity?’ page

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/update/overseas.entity.change.controller.ts [hunk 1]
```diff
@@ -73,27 +73,19 @@ export const resetChangeData = async (req: Request, appData: ApplicationData) =>
...
+export const resetDataForChange = (appData: ApplicationData) => {
   if (appData.update){
     appData.update.registrable_beneficial_owner = undefined;
   }
   appData.beneficial_owners_statement = undefined;
   appData.payment = undefined;
   return appData;
 };
-
-const existingBoMoForNoChange = async (req: Request, appData: ApplicationData) => {
```
Comment (n-r-b) [inline src/controllers/update/overseas.entity.change.controller.ts:93]: I removed this function as I noticed this was a duplicate of `retrieveBoAndMoData(req, appData)`. Also as part of the `retrieveBoAndMoData(req, appData)` function, we're resetting the arrays of BO/MO data in `appData.update.review_xyz`

File: src/controllers/update/overseas.entity.change.controller.ts [hunk 2]
```diff
@@ -35,16 +35,16 @@ export const post = async (req: Request, resp: Response, next: NextFunction) =>
...
-      appData.update.no_change = noChangeStatement === "1" ? true : false;
+    if (appData.update) {
+      appData.update.no_change = noChangeStatement === "1";
       setExtraData(session, appData);
     }
 
-    if (noChangeStatement === "1"){
-      await resetChangeData(req, appData);
+    if (noChangeStatement === "1") {
+      await resetDataForNoChange(req, appData);
```
Comment (n-r-b) [inline src/controllers/update/overseas.entity.change.controller.ts:44]: Renamed this to match what exactly we are going

File: test/controllers/update/overseas.entity.change.controller.spec.ts [hunk 3]
```diff
@@ -121,44 +128,43 @@ describe("Overseas entity do you want to change your OE controller", () => {
     });
 
     test(`redirect to ${WHO_IS_MAKING_UPDATE_URL} on YES selection`, async () => {
-      mockGetCompanyPscService.mockReturnValue(MOCK_GET_COMPANY_PSC_ALL_BO_TYPES);
-      mockGetCompanyOfficers.mockReturnValue(MOCK_GET_COMPANY_OFFICERS);
       mockGetApplicationData.mockReturnValueOnce(APPLICATION_DATA_MOCK);
-      mockGetCompanyProfile.mockReturnValueOnce(companyProfileQueryMock);
       const resp = await request(app).post(UPDATE_DO_YOU_WANT_TO_MAKE_OE_CHANGE_URL)
-        .send({ [NoChangeKey]: "1" });
+        .send({ [NoChangeKey]: "0" });
```
Comment (n-r-b) [inline test/controllers/update/overseas.entity.change.controller.spec.ts:133]: I noticed this test is for the change journey but the data being passed is for the no change journey (NoChnageKey=true), hence the updates I've made

File: src/controllers/update/overseas.entity.change.controller.ts (hunk 1 above)
Comment (AndreKramer7654) [reply src/controllers/update/overseas.entity.change.controller.ts:93]: Thanks for clarifying.

**Review by AndreKramer7654 — APPROVED** [review]

**Review by bmcgreevyCH — APPROVED** [review]

**Pull request merged**